test/secmemtest.c

```diff
@@ -75,12 +75,31 @@ static int test_sec_mem(void)
     TEST_ptr_null(OPENSSL_secure_malloc((size_t)-1));
     TEST_true(CRYPTO_secure_malloc_done());
 
+    /*-
+     * There was also a possible infinite loop when the number of
+     * elements was 1<<31, as |int i| was set to that, which is a
+     * negative number. However, it requires minimum input values:
+     *
+     * CRYPTO_secure_malloc_init((size_t)1<<34, (size_t)1<<4);
+     *
+     * Which really only works on 64-bit systems, and even then the
+     * code attempts to allocate 16 GB secure memory arena. Linux
+     * can deal with this better than other Unixy OS's (e.g. MacOS)
```

Comment:
I don't quite follow. Is suggestion that other Unixes do something else besides succeeding to mmap 16GB or failing? Or that criteria for success to mmap 16GB is different? I.e. it wouldn't be limited by available *virtual* memory? As implied I can imagine that mlock2 gives an advantage of not being limited by available *physical* memory at the moment of call (only by 'memorylocked' limit). This would make it suitable for testing like this one. If it's not about mlock2 (available since 4.4), then it sounds more like an urban legend...